src/structures/idt.rs

```diff
@@ -611,14 +614,27 @@ impl<F> Entry<F> {
     #[inline]
     fn set_handler_addr(&mut self, addr: u64) -> &mut EntryOptions {
         use crate::instructions::segmentation;
+        // SAFETY: CS segment is loaded directly from the register, so it's valid.
+        unsafe { self.set_handler_addr_unchecked(addr, segmentation::cs()) }
+    }
 
+    #[inline]
+    const unsafe fn set_handler_addr_unchecked(
+        &mut self,
+        addr: u64,
+        cs_segment: SegmentSelector,
+    ) -> &mut EntryOptions {
         self.pointer_low = addr as u16;
         self.pointer_middle = (addr >> 16) as u16;
         self.pointer_high = (addr >> 32) as u32;
 
-        self.gdt_selector = segmentation::cs().0;
+        self.gdt_selector = cs_segment.0;
+
+        // Note: we can't use `self.options.set_present` due to the underlying
+        // `set_bit` call being non-const... and that cant be const since
+        // `BitField` is a trait and traits cant have const associated methods.
+        *(&mut self.options.0) |= 1 << 15;
```

Comment:
To avoid duplicating the bit access, it would be better to move this code into the `set_present` method and make it a const fn. Also, I think this could be just `self.options.0 |= 1 << 15`.